### PR TITLE
Adds depth limit to metaphysics queries.

### LINKED ISSUE #777: Limit Query Depth
To prevent overloading Gravity servers, we should limit the depth of our queries.

### PR DESCRIPTION
**Pull request by ashfurrow**
Fixes #777. Used a depth limit of 10 as a default, and added logging so we can monitor staging for a while to see if there are any problems.

I ran through Force, Emission, Eigen, and Prediction uses of metaphysics and the largest depth I found was 7, so 10 feels like a good starting point.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -86,6 +87,7 @@ app.use(
         ...createLoaders(accessToken, userID, requestID),
       },
       formatError: graphqlErrorHandler(request.body),
+      validationRules: [depthLimit(10, { ignore: [] }, depths => console.log("Depths: ", depths))],
```
Comment (ashfurrow) [inline index.js]: This produced a JSDoc lint error, fixed here: https://github.com/stems/graphql-depth-limit/pull/1

**Comment by orta** [comment]
@alloy it's your team that's most likely to hit the limit occasionally, this can obviously be bumped if we need it - feels good to me @ash

**Comment by orta** [comment]
The JSDoc error is a part of our TypeScript https://github.com/artsy/metaphysics/pull/752 partial implementation - we don't actually fail builds for the issues, but obviously the editor showing a red error encourages our move towards TS

**Comment by joeyAghion** [comment]
10 actually seems very high to me. I'd love to start more conservatively and only bump up as necessary, but I realize this may cause problems in production. Maybe it should be configurable so we can experiment with leaving it lower in staging?

**Comment by ashfurrow** [comment]
Sure, I'll add an env var 👍

**Comment by orta** [comment]
FWIW: So far, the largest depth that's gone through staging is 5

**Comment by ashfurrow** [comment]
https://github.com/artsy/metaphysics/pull/779

**Pull request merged**